streamer/src/nonblocking/stream_throttle.rs

```diff
@@ -0,0 +1,47 @@
+use std::{
```

Comment:
can this be done without the refactor? this is stable branch. we need the absolutely most concise effective diff

Reply:
this file was not introduced by either of the PRs that this one is backporting. no tests? straight to mb?

you can try to get @CriesofCarrots or @sakridge to sign off